bin/10_3/src/Temp_stats.cpp

```diff
@@ -0,0 +1,92 @@
+#include <algorithm>
+#include <exception>
+#include <iostream>
+#include <fstream>
+#include <string>
+#include <vector>
+
+struct Reading {
+    int hour; // [0, 23].
+    double temperature; // Fahrenheit.
+};
+
+auto ReadFromFile(const std::string& fileName) ->
+std::vector<Reading> {
+    std::vector<Reading> readings;
+    std::ifstream ifs(fileName);
+    
+    if (ifs.is_open()) {
+        Reading r;
+        while (!ifs.eof()) {
+            ifs >> r.hour >> r.temperature;
+            readings.push_back(r);
+        }
+    }
+    else {
+        throw std::exception{ "Invalid file specified." };
+    }
+
+    return readings;
+}
+
+void Print(const std::vector<Reading>& readings) {
+    for (const auto& r : readings) {
+        std::cout << r.hour << " " << r.temperature << std::endl;
+    }
+    std::cout << std::endl;
+}
+
+auto AverageTemperature(const std::vector<Reading>& readings) -> 
+double {
+    double sum { 0 };
+    for (const auto& r : readings) {
+        sum += r.temperature;
+    }
+    return sum / readings.size();
+}
+
+auto TemperatureMedian(const std::vector<Reading>& readings) ->
+double {
+    std::vector<double> temperatures;
+
```

Comment:
You've forgotten about preallocating needed space.
temperatures.reserve(readings.size());